api/v1alpha1/release_types_test.go

```diff
@@ -0,0 +1,671 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ReleaseReasonFake ReleaseReason = "Fake"
+	ReleaseReasonFoo  ReleaseReason = "Foo"
+	ReleaseReasonBar  ReleaseReason = "Bar"
+)
+
+type commonArgs struct {
+	status  metav1.ConditionStatus
+	reason  ReleaseReason
+	message string
+}
+
+var _ = Describe("release_types.go test collection", func() {
+
+	Context("ReleaseReason.String()", func() {
+		It("should return a string: fake", func() {
+			Expect(ReleaseReasonFake.String()).To(Equal("Fake"),
+				"Expected \"%+v\" to equal \"%+v\"", ReleaseReasonFake.String(), "Fake")
+		})
+
+		It("should return a string: foo", func() {
+			Expect(ReleaseReasonFoo.String()).To(Equal("Foo"),
+				"Expected \"%+v\" to equal \"%+v\"", ReleaseReasonFoo.String(), "Foo")
+		})
+
+		It("should return a string: bar", func() {
+			Expect(ReleaseReasonBar.String()).To(Equal("Bar"),
+				"Expected \"%+v\" to equal \"%+v\"", ReleaseReasonBar.String(), "Bar")
+		})
+
+		It("should return an empty string", func() {
+			var ReleaseEmptyReason ReleaseReason = ""
+			Expect(ReleaseEmptyReason.String()).To(Equal(""),
+				"Expected \"%+v\" to equal \"%+v\"", ReleaseEmptyReason.String(), "")
+		})
+	})
+
+	Context("HasBeenDeployed()", func() {
+		It("should return false because status.SnapshotEnvironmentBinding is nil", func() {
+			r := &Release{}
+			Expect(r.HasBeenDeployed()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+
+		It("should return false because status.SnapshotEnvironmentBinding is empty", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					SnapshotEnvironmentBinding: "", // empty
+				},
+			}
+			Expect(r.HasBeenDeployed()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+
+		It("should return true because status.SnapshotEnvironmentBinding is not empty", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					SnapshotEnvironmentBinding: "foo",
+				},
+			}
+			Expect(r.HasBeenDeployed()).To(BeTrue(),
+				"Expected to get: %+v ; Actually Got: %+v\n", true, false)
+		})
+	})
+
+	Context("HasStarted()", func() {
+		It("should return false because Status.startTime is nil", func() {
```

Comment:
```suggestion
		It("should return false when Status.startTime is nil", func() {
```

This comment applies to the other tests as well. Basically change `because` by `when`.